contrib/database/sql/conn.go

```diff
@@ -59,43 +59,31 @@ func (tc *tracedConn) BeginTx(ctx context.Context, opts driver.TxOptions) (tx dr
 
 func (tc *tracedConn) PrepareContext(ctx context.Context, query string) (stmt driver.Stmt, err error) {
 	start := time.Now()
-	cquery, spanID := tc.withSQLCommentsInjected(ctx, query, true)
+	cquery, spanID := tc.injectComments(ctx, query, true)
 	if connPrepareCtx, ok := tc.Conn.(driver.ConnPrepareContext); ok {
 		stmt, err := connPrepareCtx.PrepareContext(ctx, cquery)
-		if spanID > 0 {
-			tc.tryTrace(ctx, queryTypePrepare, query, start, err, tracer.WithSpanID(spanID))
-		} else {
-			tc.tryTrace(ctx, queryTypePrepare, query, start, err)
-		}
+		tc.tryTrace(ctx, queryTypePrepare, cquery, start, err, tracer.WithSpanID(spanID))
```

Comment:
Wouldn't it be better to use the query without the comment in tracing? There is no point to send the comment to the agent. No provided value.
```suggestion
		tc.tryTrace(ctx, queryTypePrepare, query, start, err, spanID)
```
Also, we don't know of any situation where we'd need any other opts, so passing just the span ID should be enough.

Reply:
We had discussed offline but this was done in order to have access to the query with injected comments for testing purposes. We decided to shift to a mocked driver approach which is now included in the latest version. 

I just realized I had missed that comment about passing span id instead of options. I was going to do it as I realized I missed it but the one thing that kept me from doing that was that we've got other methods like `Begin` and `Ping` where there's no span ID to pass in since there's no sql comment injection possible. I can make those pass in a `0` but I personally prefer the optional span id to be given as an optional parameter (in this case, options). But let me know if you still want to go with the mandatory span id parameter and I'll update. 🖖